internal/client/client.go

```diff
@@ -0,0 +1,31 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type Client struct {
+	URI string
+}
+
+func NewClient(serverURI string) Client {
```

Comment:
It's enough to keep New() and when you call it somewhere you'll see client.New(). (client.NewClient() too much "client")

Reply:
Also I'll suggest to return private struck because of accesses reasons